pkg/expression/builtin_string.go

```diff
@@ -1604,24 +1604,32 @@ func (c *hexFunctionClass) getFunction(ctx BuildContext, args []Expression) (bui
 	}
 
 	argTp := args[0].GetType(ctx.GetEvalCtx()).EvalType()
+	argLen := args[0].GetType(ctx.GetEvalCtx()).GetFlen()
 	switch argTp {
 	case types.ETString, types.ETDatetime, types.ETTimestamp, types.ETDuration, types.ETJson:
 		bf, err := newBaseBuiltinFuncWithTp(ctx, c.funcName, args, types.ETString, types.ETString)
+		bf.tp.SetFlen(types.UnspecifiedLength)
 		if err != nil {
 			return nil, err
 		}
 		argFieldTp := args[0].GetType(ctx.GetEvalCtx())
 		// Use UTF8MB4 as default.
+		if argLen != types.UnspecifiedLength {
```

Comment:
I think it's better to use `argFieldTp.GetFlen()` as the `argLen` because inside `newBaseBuiltinFuncWithTp`, args[0] maybe changed by wrap some implict cast function